### PR TITLE
Bug 1156439 - [Email] Fix remaining jshint issues

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1156439

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=62183fb16b2b73c2ad085af6295767444bac0eed

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0255abab0b7668d2a74720abbd60a5725e04e7c0

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=db59acc9d259561313b1808f52b3971ad8827dc7

File: apps/email/js/html_cache_restore.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
+'use strict';
 /*jshint browser: true */
 /*global performance, console, Notification */
-'use strict';
```
Comment (jrburke) [inline apps/email/js/html_cache_restore.js]: This seems like an unnecessary change, jshint passes for me without this change, Might be good to remove it to just keep the diff smaller.

File: apps/email/test/config.js [hunk 2]
```diff
@@ -91,17 +93,16 @@
 
     if (ids) {
       req(ids, function() {
-        if (callback)
+        if (callback) {
           callback.apply(null, Array.slice(arguments));
-        if (done)
-          done();
+        }
+        done && done();
```
Comment (jrburke) [inline apps/email/test/config.js]: I would prefer to just use the `if (done) {...` form as it matches the rest of the file/usual pattern we have used for this sort of thing.

File: apps/email/test/config.js [hunk 3]
```diff
@@ -91,17 +93,16 @@
...
       }, function(err) {
-        if (done)
-          done(err);
+        done && done(err);
```
Comment (jrburke) [inline apps/email/test/config.js]: Same here.

File: apps/email/test/marionette/local_drafts_test.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+'use strict';
+
```
Comment (jrburke) [inline apps/email/test/marionette/local_drafts_test.js]: For me to run this test, Firefox complained that `const` was not allowed with that setup. So locally I changed the `const` variables in this file to `var` then it ran, so I suggest doing that.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0da5cc2c8f6fef3ff2f6883fa821df39ecc4d06e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=84fae4e7b423b448772dad0d84cb0eae49bd89c4

File: apps/email/test/marionette/local_drafts_test.js (hunk 4 above)
Comment (KevinGrandon) [inline apps/email/test/marionette/local_drafts_test.js]: Done, thanks.
